internal/provider/resource_machine.go

```diff
@@ -64,6 +64,27 @@ func resourceMachine() *schema.Resource {
 				Optional:    false,
 				Required:    false,
 			},
+			"ssh_address": {
+				Description: "The user@host directive for manual provisioning an existing machine via ssh.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				ForceNew:    true,
+			},
+			"public_key": {
+				Description: "The file path to read the public key from.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				ForceNew:    true,
```

Comment:
I'm not sure about this `ForceNew`. If a machine key is updated, this will force us to recreate it entirely. That may lead to other problems.

Reply:
The `ForceNew` is required with the Optional=true fields, but as we talked about in the meeting, this is something we should test to see how the teardown/re-create works in key update scenarios.